File: 0955-complete-binary-tree-inserter/0955-complete-binary-tree-inserter.java
class CBTInserter {
	private Queue<TreeNode> queue;
	private TreeNode root;

	public CBTInserter(TreeNode root) {
		this.root = root;
		this.queue = new LinkedList<>();
		this.queue.offer(this.root);
		while (this.queue.peek().left != null && this.queue.peek().right != null) {
			TreeNode treeNode = this.queue.poll();
			this.queue.offer(treeNode.left);
			this.queue.offer(treeNode.right);
		}
		if (this.queue.peek().left != null) {
			this.queue.offer(this.queue.peek().left);
		}
	}
	public int insert(int val) {
		TreeNode treeNode = this.queue.peek();
		if (treeNode.left == null) {
			treeNode.left = new TreeNode(val);
			this.queue.offer(treeNode.left);
		} else {
			treeNode.right = new TreeNode(val);
			this.queue.poll();
			this.queue.offer(treeNode.right);
		}
		return treeNode.val;
	}
	public TreeNode get_root() {
		return this.root;
	}
}

File: 0955-complete-binary-tree-inserter/0955-complete-binary-tree-inserter.py
class CBTInserter:
    def __init__(self, root: Optional[TreeNode]):
        self.root = root
        self.deque = deque()
        queue = deque([root])
        while queue:
            node = queue.popleft()
            if not node.left or not node.right:
                self.deque.append(node)
            if node.left:
                queue.append(node.left)
            if node.right:
                queue.append(node.right)
        
    def insert(self, val: int) -> int:
        new_node = TreeNode(val)
        parent = self.deque[0]
        if not parent.left:
            parent.left = new_node
        else:
            parent.right = new_node
            self.deque.popleft()
        self.deque.append(new_node)
        return parent.val

    def get_root(self) -> Optional[TreeNode]:
        return self.root